### PR TITLE
daktilo: init at 0.5.0

### PR DESCRIPTION
**Pull request by orhun**
## Description of changes

Add [`daktilo`](https://github.com/orhun/daktilo) 🎉

cc @matthiasbeyer (danke für die Hilfe!)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
@ofborg build daktilo

File: pkgs/tools/misc/daktilo/default.nix [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    repo = "daktilo";
+    rev = "v${version}";
+    hash = "sha256-kbJwBOUODtHdngbfa6HbbQJ0kgW6f64c0EG3y8wLymw=";
+  };
+
+  cargoHash = "sha256-u9vL2HAUgP43ZDwIEK2u/I+KUEjQsfXda03gnGJ1Krc=";
+
+  nativeBuildInputs = [
+    installShellFiles
+    pkgs.pkg-config
```
Comment (h7x4) [inline pkgs/tools/misc/daktilo/default.nix:26]: Please add `pkg-config` to the list of inputs up at the top

```suggestion
    pkg-config
```

File: pkgs/tools/misc/daktilo/default.nix [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  ];
+
+  buildInputs = with pkgs; [
+    alsa-lib
+    xorg.libX11
+    xorg.libXi
+    xorg.libXtst
```
Comment (h7x4) [inline pkgs/tools/misc/daktilo/default.nix:33]: Please add these as inputs up at the top as well

```suggestion
  buildInputs = [
    alsa-lib
    xorg.libX11
    xorg.libXi
    xorg.libXtst
```

File: pkgs/tools/misc/daktilo/default.nix [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    rm $out/bin/daktilo-{completions,mangen}
+  '';
+
+  meta = with lib; {
+    description = "Turn your keyboard into a typewriter";
+    homepage = "https://github.com/orhun/daktilo";
+    changelog = "https://github.com/orhun/daktilo/blob/${src.rev}/CHANGELOG.md";
+    license = with licenses; [ asl20 mit ];
+    maintainers = with maintainers; [ orhun ];
```
Comment (h7x4) [inline pkgs/tools/misc/daktilo/default.nix:61]: Let's set `mainProgram`

```suggestion
    maintainers = with maintainers; [ orhun ];
    mainProgram = "daktilo";
```

File: pkgs/tools/misc/daktilo/default.nix [hunk 4]
```diff
@@ -0,0 +1,63 @@
+{ lib
+, rustPlatform
+, fetchFromGitHub
+, installShellFiles
+, stdenv
+, darwin
+, unixtools
+, pkgs
```
Comment (delroth) [inline pkgs/tools/misc/daktilo/default.nix:8]: `pkgs` should not be taken as argument - instead, depend directly on the things you're importing from `pkgs`. In this case, this is just `pkg-config`.

**Review by delroth — CHANGES_REQUESTED** [review]
Overall looks good, thanks for your contribution! One small change required before this is good to merge.

File: pkgs/tools/misc/daktilo/default.nix (hunk 4 above)
Comment (delroth) [reply pkgs/tools/misc/daktilo/default.nix:8]: Ah, I missed the "with pkgs;" too - listen to @h7x4 instead, they're doing a better job at reviewing this than I am :P

**Comment by orhun** [comment]
Addressed the review comments!

**Review by delroth — APPROVED** [review]

**Comment by delroth** [comment]
@ofborg build daktilo

**Pull request merged**